src/sail_lean_backend/Sail/Sail.lean

```diff
@@ -147,4 +147,30 @@ def addInt {w : Nat} (x : BitVec w) (i : Int) : BitVec w :=
   x + BitVec.ofInt w i
 
 end BitVec
+
+namespace Nat
+
+-- NB: below is taken from Mathlib.Logic.Function.Iterate
+/-- Iterate a function. -/
+def iterate {α : Sort u} (op : α → α) : Nat → α → α
+  | 0, a => a
+  | Nat.succ k, a => iterate op k (op a)
+
+end Nat
+
+namespace Int
+
+def intAbs (x : Int) : Int := Int.ofNat (Int.natAbs x)
+
+def shiftl (a : Int) (n : Int) : Int :=
+  match n with
+  | Int.ofNat n => Sail.Nat.iterate (fun x => x * 2) n a
```

Comment:
Were you still planning on using shift operators for these? We're going to merge, but maybe you can open another PR with those updates.